src/main/java/net/snowflake/ingest/streaming/internal/ParquetRowBuffer.java

```diff
@@ -91,20 +94,54 @@ public void setupSchema(List<ColumnMetadata> columns) {
       if (!column.getNullable()) {
         addNonNullableFieldName(column.getInternalName());
       }
-      this.statsMap.put(
-          column.getInternalName(),
-          new RowBufferStats(column.getName(), column.getCollation(), column.getOrdinal()));
-
-      if (onErrorOption == OpenChannelRequest.OnErrorOption.ABORT
-          || onErrorOption == OpenChannelRequest.OnErrorOption.SKIP_BATCH) {
-        this.tempStatsMap.put(
+      if (!clientBufferParameters.getIsIcebergMode()) {
+        /* Streaming to FDN table doesn't support sub-columns, set up the stats here. */
+        this.statsMap.put(
             column.getInternalName(),
-            new RowBufferStats(column.getName(), column.getCollation(), column.getOrdinal()));
+            new RowBufferStats(
+                column.getName(), column.getCollation(), column.getOrdinal(), null /* fieldId */));
+
+        if (onErrorOption == OpenChannelRequest.OnErrorOption.ABORT
+            || onErrorOption == OpenChannelRequest.OnErrorOption.SKIP_BATCH) {
+          this.tempStatsMap.put(
+              column.getInternalName(),
+              new RowBufferStats(
+                  column.getName(),
+                  column.getCollation(),
+                  column.getOrdinal(),
+                  null /* fieldId */));
+        }
       }
 
       id++;
     }
     schema = new MessageType(PARQUET_MESSAGE_TYPE_NAME, parquetTypes);
+    if (clientBufferParameters.getIsIcebergMode()) {
+      /* Iceberg mode requires stats for sub-columns, set them up here. */
+      for (ColumnDescriptor subColumnDescriptor : schema.getColumns()) {
```

Comment:
nit: columnDescriptor, since its not just subcolumns that yo're processing here.

Reply:
Renamed.